lib/src/method_channel.dart

```diff
@@ -43,32 +46,45 @@ class ShowMenuArgs {
   }
 }
 
-const _channel = MethodChannel('native_context_menu');
+final _channel = const MethodChannel('native_context_menu')
+  ..setMethodCallHandler(
+    (call) async {
+      switch (call.method) {
+        case 'id':
+          {
+            // Notify about the selected menu item's id & complete the future.
+            _completer.complete(call.arguments);
+            break;
+          }
+        default:
+          break;
+      }
+    },
+  );
+Completer<int> _completer = Completer<int>();
 int _menuItemId = 0;
 
 Future<MenuItem?> showContextMenu(ShowMenuArgs args) async {
   final menu = _buildMenu(args.items);
   _menuItemId = 0;
-
-  final id = await _channel.invokeMethod('showMenu', args.toJson());
-
+  _channel.invokeMethod('showMenu', args.toJson());
+  int id = await _completer.future;
+  _completer = Completer<int>();
   if (id != -1) {
     return menu[id];
   }
 }
 
 Map<int, MenuItem> _buildMenu(List<MenuItem> items) {
   final built = <int, MenuItem>{};
-
```

Comment:
newlines is a matter of preference, don't drop these just because you personally don't like them 🙂